src/Data/Yaml.hs

```diff
@@ -100,65 +106,89 @@ import qualified Data.Text.Lazy as TL
 import Data.Text.Lazy.Builder (toLazyText)
 import qualified Data.Vector as V
 import System.IO.Unsafe (unsafePerformIO)
+import Data.Text (Text)
 
 import Data.Yaml.Internal
-import Text.Libyaml hiding (encode, decode, encodeFile, decodeFile)
+import Text.Libyaml hiding (encode, decode, encodeFile, decodeFile, encodeWith, encodeFileWith)
 import qualified Text.Libyaml as Y
 
+data EncodeOptions = EncodeOptions
+  { encodeOptionsStringStyle :: Text -> ( Tag, Style )
```

Comment:
For whatever identifier gets exported providing support for this, there should be a public comment explaining that missetting this can result in semantically-mismatched YAML documents due to #24. Also, please ensure that any exposed identifiers have a Haddock comment with a `@since` comment.